src/seedu/addressbook/data/person/UniquePersonList.java

```diff
@@ -37,6 +37,8 @@ protected DuplicatePersonException() {
 
     private final List<Person> internalList = new ArrayList<>();
 
+    private final List<Person> groupedPersonList = new ArrayList<>();
```

Comment:
Perhaps add comments to `groupedPersonList` and explain what it does and why do we need this?